File: Oxiyopal.Models/Ciudad.cs
using Oxiyopal.Models.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace Oxiyopal.Models
{
    public class Ciudad : EntityBase<int>
    {
        public string NombreCiudad { get; set; }

        public int DepartamentoId { get; set; }
        public Departamento Departamento { get; set; }
        public IList<Bodega> Bodegas { get; set; }
        public IList<Cliente> Clientes { get; set; }
    }
}

File: Oxiyopal.WebUI/ClientApp/src/app/models/Ciudad.ts
import { EntityBase } from './Base/EntityBase';
import { Cliente } from './Cliente';
import { Bodega } from './Bodega';
import { Departamento } from './Departamento';

export interface Ciudad extends EntityBase<number> {
    nombreCiudad: string;

    departamentoId: number;
    departamento: Departamento;
    bodegas: Bodega[];
    clientes: Cliente[];
}

